include/aspect/simulator.h

```diff
@@ -1194,6 +1197,27 @@ namespace aspect
       check_consistency_of_formulation ();
 
       /**
+       * Computes the initial Newton residual.
+       */
+      double
+      compute_initial_newton_residual (LinearAlgebra::BlockVector &linearized_stokes_initial_guess);
+
+      /**
+       * This function computes the Eisenstat Walker linear tolerance used for the Newton Stokes solver.
+       * The Eisenstat and Walker (1996) method is used for determining the linear tolerance of
```

Comment:
Yes, it refers to the 1996 paper (Choosing the forcing terms in an inexact Newton method). How do you want me to reference it exactly here? Is just adding the doi (http://dx.doi.org/10.1137/0917003) enough?